lib/matplotlib/artist.py

```diff
@@ -844,6 +827,30 @@ def get_in_layout(self):
         """
         return self._in_layout
 
+    def _get_skip_layout(self):
```

Comment:
Maybe call this `_fully_clipped_to_axes` or something similarly more descriptive as to _why_ we want to skip it in the layout.  There are many reasons a user could imagine you want to skip the including an Artists in the layout, we want to avoid the case where each user imagines that this method implements _their_ favorite justification ;)

The name should also hint that this only matters for Axes and not for figure level layouts.

Reply:
Done!